fda/basis.py

```diff
@@ -1778,6 +1784,21 @@ def copy(self, *, basis=None, coefficients=None, dataset_label=None,
                           axes_labels=axes_labels, extrapolation=extrapolation,
                           keepdims=keepdims)
 
+    def inprod(self, other):
+        if self.domain_range != other.domain_range:
+            raise ValueError("Both Objects should have the same domain_range")
+        if isinstance(other, Basis):
+            other = other.to_basis()
+        inprodmat = [
+            scipy.integrate.quad(lambda x: self[i].evaluate([x]) * other[j].evaluate([x]), self.domain_range[0],
+                                 self.domain_range[1])[0] for j in range(0, other.nsamples) for i in
+                                 range(0, self.nsamples)]
+        return numpy.array(inprodmat).reshape((self.nsamples, other.nsamples))
+
+    def __getitem__(self, given):
```

Comment:
The function `__getitem__` was added in the pull request   #26 `feature/FData`.

Reply:
I discard it to use that one

Reply:
In the new version the attributes are not being preserved and they'll have the default values in the slice.

```
>>> fdbasis.extrapolation = "zeros"
>>> fdbasis.keepdims = True
>>> fdbasis
FDataBasis(
    basis=Monomial(domain_range=[array([0, 1])], nbasis=1),
    coefficients=[[0.5]
     [1. ]],
    dataset_label=None,
    axes_labels=None,
    extrapolation=<fda.extrapolation.FillExtrapolation object at 0x1a19846b00>,
    keepdims=True)
>>> fdbasis[0]
FDataBasis(
    basis=Monomial(domain_range=[array([0, 1])], nbasis=1),
    coefficients=[[0.5]],
    dataset_label=None,
    axes_labels=None,
    extrapolation=None,
    keepdims=False)

```
In the discarded version it was used the method copy to avoid this problem.
```
    return self.copy(coefficients=self.coefficients[given])
```
The case in which the key is an integer was treated separately in the discarded version, as in `FDataGrid`, due to this behaviour of the slice of numpy
```
>>> matrix.shape
(2, 3, 2)
>>> matrix[0].shape
(3, 2)
>>> matrix[0:1].shape
(1, 3, 2)
```
It is true that it is not necessary to do it currently and could be deleted, but if more dimensions are added it will be necessary to do it.

Reply:
Then only the last one is being used. Only one of them should be kept.